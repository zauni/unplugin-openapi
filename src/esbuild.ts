import { unplugin } from "./index.js";

export default unplugin.esbuild as typeof unplugin.esbuild;
