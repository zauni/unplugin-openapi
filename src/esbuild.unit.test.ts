import { build } from "esbuild";
import { describe, expect, it } from "vitest";
import esbuildPlugin from "./esbuild.js";

describe("esbuild plugin", () => {
	it("should handle a YAML schema with a reference", async () => {
		const result = await build({
			entryPoints: ["./src/fixtures/esbuild/main.ts"],
			bundle: true,
			plugins: [esbuildPlugin()],
			write: false,
		});

		const code = result.outputFiles.map((f) => f.text).join("");

		expect(code).toMatchSnapshot();
		expect(code).toContain('type: "object",');
	});

	it("should handle a JSON schema with a reference", async () => {
		const result = await build({
			entryPoints: ["./src/fixtures/esbuild/main-json.ts"],
			bundle: true,
			plugins: [
				esbuildPlugin({
					extensions: [".openapi.json"],
				}),
			],
			write: false,
		});

		const code = result.outputFiles.map((f) => f.text).join("");

		expect(code).toMatchSnapshot();
		expect(code).toContain('type: "object",');
	});
});
