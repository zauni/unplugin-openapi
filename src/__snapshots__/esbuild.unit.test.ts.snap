// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`esbuild plugin > should handle a JSON schema with a reference 1`] = `
""use strict";
(() => {
  // src/fixtures/api.openapi.json
  var api_openapi_default = {
    openapi: "3.0.0",
    info: {
      title: "My great API",
      description: "Great description",
      version: "1.0.0"
    },
    paths: {
      "/my/path": {
        get: {
          summary: "Some GET request",
          responses: {
            "200": {
              description: "Some response",
              content: {
                "application/json": {
                  schema: {
                    type: "object",
                    properties: {
                      someKey: {
                        type: "string"
                      }
                    },
                    required: [
                      "someKey"
                    ]
                  },
                  example: {
                    someKey: "some value"
                  }
                }
              }
            }
          }
        }
      }
    }
  };

  // src/fixtures/esbuild/main-json.ts
  console.log(api_openapi_default);
})();
"
`;

exports[`esbuild plugin > should handle a YAML schema with a reference 1`] = `
""use strict";
(() => {
  // src/fixtures/api.yaml
  var api_default = {
    openapi: "3.0.0",
    info: {
      title: "My great API",
      description: "Great description",
      version: "1.0.0"
    },
    paths: {
      "/my/path": {
        get: {
          summary: "Some GET request",
          responses: {
            "200": {
              description: "Some response",
              content: {
                "application/json": {
                  schema: {
                    type: "object",
                    properties: {
                      someKey: {
                        type: "string"
                      }
                    },
                    required: [
                      "someKey"
                    ]
                  },
                  example: {
                    someKey: "some value"
                  }
                }
              }
            }
          }
        }
      }
    }
  };

  // src/fixtures/esbuild/main.ts
  console.log(api_default);
})();
"
`;
